File: app/src/main/java/com/zebra/barcodeintellgencetools/RetrieveAPITask.java
package com.zebra.barcodeintellgencetools;


import android.os.AsyncTask;

import com.zebra.savanna.CreateBarcode;
import com.zebra.savanna.FDARecall;
import com.zebra.savanna.Rotation;
import com.zebra.savanna.Symbology;
import com.zebra.savanna.UPCLookup;

import org.json.JSONException;
import org.json.JSONObject;

class RetrieveAPITask extends AsyncTask<String, Void, Object> {
    private Exception exception;

    protected Object doInBackground(String... args) {
        try {
            switch (args[0]) {
                case "create":
                    return CreateBarcode.create(Symbology.fromString(args[2]), args[1], ItemListActivity.density, Rotation.Normal,true);
                case "lookup":
                    return UPCLookup.lookup(args[1]);
                case "deviceSearch":
                    return FDARecall.deviceSearch(args[1]);
                case "drugSearch":
                    return FDARecall.drugSearch(args[1]);
                case "drugUpc":
                    return FDARecall.drugUpc(args[1]);
                case "foodUpc":
                    return FDARecall.foodUpc(args[1]);
            }
        } catch (Exception e) {
            this.exception = e;
        }
        return null;
    }

    protected void onPostExecute(Object apiData) {
        if (exception != null) {
            try {
                ItemDetailFragment.onPostExecute(new JSONObject(exception.getMessage()));
            } catch (JSONException e) {
                e.printStackTrace();
            }

        } else {
            ItemDetailFragment.onPostExecute(apiData);
        }
    }
}

File: BarcodeIntelligenceTools/API/RetrieveAPITask.cs
using Android.OS;
using Savanna;
using System;

namespace BarcodeIntelligenceTools.API
{
    public class RetrieveAPITask : AsyncTask<string, Java.Lang.Void, object>
    {
        private Exception exception;

        protected override object RunInBackground(params string[] args)
        {
            try
            {
                switch (args[0])
                {
                    case "create":
                        return CreateBarcode.CreateAsync(Enum.Parse<Symbology>(args[2].Replace('-', '_')), args[1], ItemListActivity.density, Rotation.Normal, true);
                    case "lookup":
                        return UPCLookup.LookupAsync(args[1]);
                    case "deviceSearch":
                        return FDARecall.DeviceSearchAsync(args[1]);
                    case "drugSearch":
                        return FDARecall.DrugSearchAsync(args[1]);
                    case "drugUpc":
                        return FDARecall.DrugUpcAsync(args[1]);
                    case "foodUpc":
                        return FDARecall.FoodUpcAsync(args[1]);
                }
            }
            catch (Exception e)
            {
                this.exception = e;
            }
            return null;
        }

        protected override void OnPostExecute(object apiData)
        {
            if (exception != null)
            {
                ItemDetailFragment.Instance.onPostExecute(exception.Message);
            }
            else
            {
                ItemDetailFragment.Instance.onPostExecute(apiData);
            }
        }
    }
}